### PR TITLE
Project gms

### PR DESCRIPTION
**Pull request by SnowNooDLe**
# Projects tab - GMS

Projects GMS is implemented, Tooltips need to be updated also the disclaimer? at some stage.

image

image

image

image

image

## .gitignore update - requirements.txt

As we recently changed the workflow of the branches, master_dev and master_ea

master_dev treats as a playground, when a feature seems working, just merge it into master_dev and deploy it to the psha-dev.

Then once it looks good and ready for the review, open PR to master_ea and get a code review.

Hence, the psha-dev will be based on the `master_dev` and the psha-ea will be based on the `master_ea`. Hence the middleware package will be installed from a different branch depends on the deployed branch:
- psha-dev's middleware will be installed from master_dev
- psha-ea's middleware will be installed from master_ea

### CODE REVIEW

File: frontend/src/components/Project/GMS/GMSViewer.js [hunk 1]
```diff
@@ -1,7 +1,305 @@
...
+
+    const getGMSData = async () => {
+      if (projectGMSGetClick !== null) {
+        try {
+          setShowErrorMessage({ isError: false, errorCode: null });
+          setIsLoading(true);
+
+          const token = await getTokenSilently();
+
+          let queryString = `?project_id=${projectId}&station_id=${projectLocationCode[projectLocation]}_${projectVS30}&gms_id=${projectGMSId}`;
```
Comment (jasonmotha) [inline frontend/src/components/Project/GMS/GMSViewer.js:124]: Is there not a query builder? Seems odd that you have to build it manually?

**Review by jasonmotha — DISMISSED** [review]

File: frontend/src/components/Project/GMS/GMSViewer.js [hunk 2]
```diff
@@ -1,7 +1,305 @@
...
+
+  const [causalParamBounds, setCausalParamBounds] = useState({});
+
+  const [showErrorMessage, setShowErrorMessage] = useState({
+    isError: false,
+    errorCode: null,
+  });
+
+  // Reset the state to prevent auto-trigger
+  // E.g. change tab between Pojrects and Hazard Analyis
```
Comment (jasonmotha) [inline frontend/src/components/Project/GMS/GMSViewer.js:62]: Pojrects -> rekt

**Review by claudio525 — DISMISSED** [review]

**Review by claudio525 — DISMISSED** [review]

**Review by jasonmotha — CHANGES_REQUESTED** [review]
Make a function

File: frontend/src/components/common/GMS/GMSViewerCausalParameters.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ const GMSViewerCausalParameters = ({
   const boundsRangeY = [0, 1];
 
   // X-axis range - Default
-  let xAxisRange = [Math.min(...xRange) * 0.9, Math.max(...xRange) * 1.1];
+  let xAxisRange = [-1, Math.max(...xRange) * 1.1];
```
Comment (jasonmotha) [inline frontend/src/components/common/GMS/GMSViewerCausalParameters.js:43]: xtr_space = Math.max(...xRange) * 0.1
xRange = (0 - xtr_space, xtr_space * 11) 

This should make the spacing on both sides equal? Assuming the axis is linear space.

File: frontend/src/components/common/GMS/GMSViewerCausalParameters.js [hunk 4]
```diff
@@ -103,10 +103,16 @@ const GMSViewerCausalParameters = ({
...
 
-    xAxisRange = [
-      Math.min(...xRange, causalParamBounds[metadata]["min"]) * 0.9,
-      Math.max(...xRange, causalParamBounds[metadata]["max"]) * 1.1,
-    ];
+    let minXCoord = Math.min(...xRange, causalParamBounds[metadata]["min"]);
+    let maxXCoord = Math.max(...xRange, causalParamBounds[metadata]["max"]);
+
+    xAxisRange =
+      metadata === "mag"
```
Comment (jasonmotha) [inline frontend/src/components/common/GMS/GMSViewerCausalParameters.js:110]: Comment to explain the cases here

**Comment by SnowNooDLe** [comment]
Close due to more changes to come & prevent sending a notification every time I push.

**Pull request closed without merging**